src/slice.jl

```diff
@@ -71,14 +72,8 @@ end
 
 function Base.insert!(s::Slice{T, N}, i::Integer, value) where {T, N}
     if s.n < N && 1 <= i <= s.n + 1
-        # TODO: This should work but won't. Fix this in Julia.
-        #copy!(s.data, i+1, s.data, i, s.n - i + 1)
-        if isbits(T)
-            unsafe_copy!(pointer(s.data, i+1), pointer(s.data, i), s.n - i + 1)
-        else
-            for k in 0:(s.n - i)
-                @inbounds s.data[s.n+1-k] = s.data[s.n-k]
-            end
+        for k in 0:(s.n - i)
+            @inbounds s.data[s.n+1-k] = s.data[s.n-k]
```

Comment:
Why did you remove the fast path (using unsafe_copy! for isbits types)?

Reply:
I found it rather slowing down the test runs and allocating 1K bytes per each allocation and 10+ms for every execution. The original test case pass time was coming down 50 times slower than 0.6. Hence, had to eliminate additional allocations and calls that were contributing to that.

Reply:
Had you tried changing the name to `unsafe_copyto!`?
The slowdown may just have been because of the deprecation of `unsafe_copy!`
(deprecated code, even though they print out the deprecation warning only once, slow down the code substantially for all calls)

Reply:
Will try that. My normal interpretation of deprecation has been it's will not be maintained so don't use but if used it would do no harm. But I see in Julia many places deprecated code gets so much performance hit that it gets to the extent of being unusable.

Reply:
https://github.com/BioJulia/IntervalTrees.jl/pull/37/commits/bfe59b002e5f687d99faf5a65c1435efface8379 has the changes as suggested.